handlers/meme.js

```diff
@@ -0,0 +1,110 @@
+
+  // Instead of just throwing a console error or timing out of the values aren't set, we can send a message and then quit
+  if(!MEME_USER){
+    LOG.error('Meme generator username not set!');
+    channel.send('Meme generator username not set!');
+    return;
+  } else if(!MEME_PASSWORD){
+    LOG.error('Meme generator password not set!');
+    channel.send('Meme generator password not set!');
+    return;
+  } else if(!MEMEGENERATOR_API_KEY){
+    LOG.error('Meme generator API key not set!');
+    channel.send('Meme generator API key not set!');
+    return;
+  }
+
+  // Pulling messages from the channel to populate the meme
+  //     Limiting search to the last 100 messages (the max for the fetchMessages() function)
+  channel.fetchMessages({ before: message.id, limit: 100})
+    .then(messages => {
+      // Setting the new top and bottom text for the macro
+      top_text = findMessage(channel, messages).content;
+      bottom_text = findMessage(channel, messages).content;
+    })
+    .catch(error => {
+      // For some reason, if we run into an issue when finding the messages, we will let the user know and then use our fallbacks
+      LOG.error(error);
+      channel.send("Error getting messages for channel. Using default.");
+    });
+
+  // Before we make the macro, we need to query the API and find a generator to use.
+  //     The meme "generator" is actually the background image that is used for the macro.
+  //     We query the API for the current top 25 most popular generator images (25 is the max for the results page size),
+  //     and then randomly choose one of those to use.
+  let generatorOptions = {
+    uri: `http://version1.api.memegenerator.net//Generators_Select_ByPopular?pageIndex=0&pageSize=25&days=&apiKey=${MEMEGENERATOR_API_KEY}`,
+    json: true
+  };
+  rp.get(generatorOptions).then(generators => {
+    generatorID = _.sample(generators.result).generatorID;
+
+    // Now that we've gotten a generator, we can send our options to the meme forge and build our dark creation
+    let uri = `${API_URL}?apiKey=${MEMEGENERATOR_API_KEY}&generatorID=${generatorID}&languageCode=${languageCode}&text0=${top_text}&text1=${bottom_text}&username=${MEME_USER}&password=${MEME_PASSWORD}`;
+    let options = {
+      uri: uri,
+      json: true
+    };
+    rp.get(options)
+      .then(meme => {
+        // The API will spit back a bunch of stuff, namely the URL of the macro it just made
+        channel.send(meme.result.instanceImageUrl);
+      })
+      .catch(error =>{
+        // This will catch when we are able to query the generator, but the macro creation fails for some reason
+        //     As a fun treat, we can still send the top and bottom text of our meme-to-be to the channel
+        LOG.error(error);
+        channel.send("Encountered an error while connecting to the world meme database; we've been set up!");
+        channel.send(_.toUpper(top_text));
+        channel.send(_.toUpper(bottom_text));
```

Comment:
Probably don't need to send this if we fail?

Reply:
Better to send back a message if the API call fails; otherwise it just makes it look like the bot is stuck or encountered an unknown error. This will at least tell the channel that the bot is working fine